tests/dhcp_relay/test_dhcp_relay.py

```diff
@@ -112,12 +118,31 @@ def validate_dut_routes_exist(duthosts, rand_one_dut_hostname, dut_dhcp_relay_da
         assert len(rtInfo["nexthops"]) > 0, "Failed to find route to DHCP server '{0}'".format(dhcp_server)
 
 
-def test_dhcp_relay_default(duthosts, rand_one_dut_hostname, ptfhost, dut_dhcp_relay_data, validate_dut_routes_exist):
+def set_dual_tor(duthost):
```

Comment:
suggest making this as a fixture and use `yield` to separate the setup from teardown. Also, the new fixture can be parameterized with dhcp_run_mode params similar to [`ipVersion`](https://github.com/Azure/sonic-mgmt/blob/master/tests/arp/test_neighbor_mac_noptf.py#L45)

Reply:
wondering how to do that? each individual test runs on a random dut host which is unknown before test gets executed

Reply:
The dut will not change as [`rand_one_dut_hostname`](https://github.com/Azure/sonic-mgmt/blob/master/tests/conftest.py#L202) has module scope.

Reply:
That makes code harder to understand. I prefer per-test parameter.

Reply:
The param is going to cascade down for every test case, however you will have all single ToR dhcp test cases run in a row followed by dual ToR DHCp test cases.

Reply:
ok, if the rand_one_dut_hostname is always going to have module scope, moving setup and teardown to a fixture will be good, and rand_one_dut_hostname  will be moved into that fixture as well. I'll update the code

Reply:
updated